From 27eb6829d288e641c50cea615e91acf7e0ee45a6 Mon Sep 17 00:00:00 2001
From: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
Date: Wed, 20 Sep 2023 11:01:20 +0530
Subject: [PATCH] remoteproc: add infrastructure to build non-secure PIL

On emulation platforms, Q6 will be brought up by using
non-secure PIL. This patch adds support to build non-
secure reset sequence.

Change-Id: I8aca44f0e78abb3ebde761b7bdf8a1cd28c69452
Signed-off-by: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
---
 drivers/remoteproc/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/remoteproc/Kconfig b/drivers/remoteproc/Kconfig
index 6238af89ee5b..6a910d0f53f4 100644
--- a/drivers/remoteproc/Kconfig
+++ b/drivers/remoteproc/Kconfig
@@ -254,6 +254,15 @@ config QCOM_Q6V5_MPD
 	  for the Hexagon based MultiPD model remote processors on e.g. IPQ5018.
 	  This is trustZone wireless subsystem.
 
+config QCOM_NON_SECURE_PIL
+	tristate "Qualcomm Non secure Peripheral Image Loader"
+	depends on OF && ARCH_QCOM
+	depends on QCOM_Q6V5_MPD
+	help
+	  Say y here to support the Qualcomm Non Secure Peripheral Image Loader
+	  for the Hexagon based MultiPD model remote processors on emulation
+	  platforms.
+
 config QCOM_Q6V5_WCSS
 	tristate "Qualcomm Hexagon based WCSS Peripheral Image Loader"
 	depends on OF && ARCH_QCOM
-- 
2.34.1

